staging/src/k8s.io/api/flowcontrol/v1alpha1/types.go

```diff
@@ -0,0 +1,232 @@
+	// +optional
+	Status PriorityLevelConfigurationStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PriorityLevelConfigurationList is a list of PriorityLevelConfiguration objects.
+type PriorityLevelConfigurationList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// Items is a list of request-priorities.
+	Items []PriorityLevelConfiguration `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// PriorityLevelConfigurationSpec is specification of a priority level
+type PriorityLevelConfigurationSpec struct {
+	// GlobalDefault specifies whether this priority level should be considered as the default priority for requests
+	// that do not have any priority level. Only one PriorityClass should be marked as `globalDefault`.
+	// +optional
+	GlobalDefault bool `json:"globalDefault,omitempty" protobuf:"bytes,1,opt,name=globalDefault"`
+	// Exempt defines whether the priority level is exempted or not.  There should be at most one exempt priority level.
+	// Being exempt means that requests of that priority are not subject to concurrency limits (and thus are never queued)
+	// and do not detract from the concurrency available for non-exempt requests. In a more sophisticated system, the
+	// exempt priority level would be the highest priority level. The field is default to false and only those system
+	// preset priority level can be exempt.
+	// +optional
+	Exempt bool `json:"exempt,omitempty" protobuf:"varint,2,opt,name=exempt"`
+	// AssuredConcurrencyShares is a positive number for a non-exempt priority level, representing the weight by which
+	// the priority level shares the concurrency from the global limit. The concurrency limit of an apiserver is divided
+	// among the non-exempt priority levels in proportion to their assured concurrency shares. Basically this produces
+	// the assured concurrency value (ACV) for each priority level:
+	//
+	//             ACV(l) = ceil( SCL * ACS(l) / ( sum[priority levels k] ACS(k) ) )
+	//
+	AssuredConcurrencyShares int32 `json:"assuredConcurrencyShares,omitempty" protobuf:"varint,3,opt,name=assuredConcurrencyShares"`
+	// Queues is a number of queues that belong to a non-exempt PriorityLevelConfiguration object. The queues exist independently
+	// at each apiserver.
+	Queues int32 `json:"queues,omitempty" protobuf:"varint,4,opt,name=queues"`
+	// HandSize is a small positive number for applying shuffle sharding. When a request arrives at an apiserver the
+	// request flow identifier’s string pair is hashed and the hash value is used to shuffle the queue indices and deal
+	// a hand of the size specified here. If empty, the hand size will the be set to 1 at which  the priority drains the
```

Comment:
Omit everything after "the hand size will be set to 1".

Maybe specify that the hand size must be less than the number of queues. Should we cap it at 1/2 the number of queues? There must be some value at which increasing the hand size makes the isolation worse, but I don't know where.

Reply:
Yes, when there are many queues it is bad to have the hand size be a non-small fraction of the number of queues.  But I do not think we should try to pin down a particular formula here.  I think it would be enough to say that the hand size can not be larger than the number of queues.

Reply:
i updated and trim'd the doc.